document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -102,6 +110,28 @@ public static String prepareCast(String field, Type type) {
     }
   }
 
+  public static String prepareCastForFieldAccessor(String field, Object value) {
```

Comment:
We are using the field accessor of jsonb ethe lement in `group by` clause. So, it needs to same in  `having` clause. So, for string type, we need different casting.